src/pages/NewGroupPage.js

```diff
@@ -0,0 +1,271 @@
+    displayName: PropTypes.string.isRequired,
+});
+
+const propTypes = {
+    // All of the personal details for everyone
+    personalDetails: PropTypes.objectOf(personalDetailsPropTypes).isRequired,
+
+    // All reports shared with the user
+    reports: PropTypes.shape({
+        reportID: PropTypes.number,
+        reportName: PropTypes.string,
+    }).isRequired,
+
+    // Session of currently logged in user
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }).isRequired,
+
+    // Report ID currently in view
+    currentlyViewedReportID: PropTypes.string,
+};
+
+const defaultProps = {
+    currentlyViewedReportID: '',
+};
+
+class NewGroupPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.toggleOption = this.toggleOption.bind(this);
+        this.createGroup = this.createGroup.bind(this);
+
+        const {
+            recentReports,
+            personalDetails,
+            userToInvite,
+        } = getNewGroupOptions(
+            props.reports,
+            props.personalDetails,
+            '',
+            [],
+        );
+
+        this.state = {
+            searchValue: '',
+            recentReports,
+            personalDetails,
+            selectedOptions: [],
+            userToInvite,
+        };
+    }
+
+    /**
+     * Helper method that returns the text to be used for the header's message and title (if any)
+     *
+     * @param {Boolean} maxParticipantsReached
+     * @return {String}
+     */
+    getHeaderTitleAndMessage(maxParticipantsReached) {
+        if (maxParticipantsReached) {
+            return {
+                headerTitle: 'Maximum participants reached',
+                headerMessage: 'You\'ve reached the maximum number of participants for a group chat.',
+            };
+        }
+
+        const hasSelectableOptions = this.state.personalDetails.length + this.state.recentReports.length !== 0;
+        if (!hasSelectableOptions && !this.state.userToInvite) {
+            return {
+                headerTitle: '',
+                // eslint-disable-next-line max-len
+                headerMessage: 'Don\'t see who you\'re looking for? Type their email or phone number to invite them to chat.',
+            };
+        }
+
+        return {
+            headerTitle: '',
+            headerMessage: '',
+        };
+    }
+
+    /**
+     * Returns the sections needed for the OptionsSelector
+     *
+     * @param {Boolean} maxParticipantsReached
+     * @returns {Array}
+     */
+    getSections(maxParticipantsReached) {
+        const sections = [];
+        sections.push({
+            title: undefined,
+            data: this.state.selectedOptions,
+            shouldShow: true,
+            indexOffset: 0,
+        });
+
+        if (maxParticipantsReached) {
+            return sections;
+        }
+
+        sections.push({
+            title: 'RECENTS',
```

Comment:
Is `Recents` correct? Sounds like it shouldn't be pluralized if it's an adjective?

Reply:
Hmm I'm not sure in this case just going off of what's in the doc, but you are correct about it being an adjective.

Reply:
I'm thinking we can always change this up if people find it strange and there isn't any need to block on this.